cli/crates/backend/src/api/environment_variables/list.rs

```diff
@@ -0,0 +1,62 @@
+use common::{consts::PROJECT_METADATA_FILE, environment};
+use cynic::{http::ReqwestExt, QueryBuilder};
+use tokio::fs::read_to_string;
+
+use crate::api::{
+    client::create_client,
+    consts::api_url,
+    errors::ApiError,
+    graphql::queries::list_env_vars::{
+        ListEnvironmentVariables, ListEnvironmentVariablesArguments, ListEnvironmentVariablesBySlugs,
+        ListEnvironmentVariablesBySlugsArguments, Node, Project,
+    },
+    types::ProjectMetadata,
+};
+
+pub(super) async fn with_linked_project() -> Result<Project, ApiError> {
+    let project = environment::Project::get();
+
+    let project_metadata_file_path = project.dot_grafbase_directory_path.join(PROJECT_METADATA_FILE);
+
+    match project_metadata_file_path.try_exists() {
+        Ok(true) => {}
+        Ok(false) => return Err(ApiError::UnlinkedProject),
+        Err(error) => return Err(ApiError::ReadProjectMetadataFile(error)),
+    }
+
+    let project_metadata_file = read_to_string(project_metadata_file_path)
+        .await
+        .map_err(ApiError::ReadProjectMetadataFile)?;
+
+    let project_metadata: ProjectMetadata =
+        serde_json::from_str(&project_metadata_file).map_err(|_| ApiError::CorruptProjectMetadataFile)?;
```

Comment:
The code up to here looks like it should be somewhere more generic. But all fine if it's the first implementation.